x-pack/plugins/secops/server/utils/build_query/fields.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FieldNode, SelectionNode, SelectionSetNode } from 'graphql';
+import { isEmpty } from 'lodash/fp';
+
+export const getFields = (
+  data: SelectionSetNode | FieldNode,
+  fields: string[] = [],
+  postFields: string[] = []
+): string[] => {
+  if (data.kind === 'Field' && data.selectionSet && !isEmpty(data.selectionSet.selections)) {
+    return getFields(data.selectionSet, fields);
+  } else if (data.kind === 'SelectionSet') {
+    return data.selections.reduce(
```

Comment:
To reduce the cyclomatic complexity of this function, perhaps this `reduce` can be extracted into a named function? That might make it easier to test and read.

Reply:
I will add unit test for this like that it will be easier to understand what's going on